packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1682,7 +1682,7 @@ function detachFiberAfterEffects(fiber: Fiber) {
   // tree, which has its own pointers to children, parents, and siblings.
   // The other host nodes also point back to fibers, so we should detach that
   // one, too.
-  if (fiber.tag === HostComponent) {
+  if (fiber.tag === HostComponent || fiber.tag === HostPortal) {
```

Comment:
I think this will work but technically Instance is a different type from Container in the host config. So a more accurate approach would be to create a separate branch for HostPortal (and probably HostRoot?)  and call `detachDeletedContainer` in that branch.

See here:

https://github.com/facebook/react/blob/bfb9cbd8ca2d042677056110887cc62d438f665f/packages/react-dom-bindings/src/client/ReactDOMComponentTree.js#L52-L60

@sebmarkbage does that sound right? For context, someone at Meta reported a memory leak and one of the links in the retain chain was `'__reactListeners$' + randomKey`, which gets added to portal containers. I think it's a similar case to the other leaks where the root cause is something else in userspace, but the consequences of the leak are worse because of the backpointer into the fiber tree.